test/test_value.h

```diff
@@ -0,0 +1,36 @@
+#include <cxxtest/TestSuite.h>
+#include <cstdint>
+#include "value.h"
+
+using namespace nbt;
+
+class value_assignment_test : public CxxTest::TestSuite
+{
+public:
+    void test_numeric_assignments()
+    {
+        value v;
+
+        v = int8_t(-5);
+        TS_ASSERT_EQUALS(int32_t(v), int32_t(-5));
+        TS_ASSERT_EQUALS(double(v), static_cast<double>(int8_t(-5)));
```

Comment:
is the double casting needed? 
like you cast to int8_t then to double. can't this be done without any casting?
here is a suggestion for this line that you can apply to the others(at least for the lines with static cast):

```suggestion
        TS_ASSERT_EQUALS(double(v), 5.);
```
or

```suggestion
        TS_ASSERT_EQUALS(double(v), double(-5));
```